subsys/pm/pm.c

```diff
@@ -210,12 +210,11 @@ bool pm_system_suspend(int32_t ticks)
 	}
 
 #if defined(CONFIG_PM_DEVICE) && !defined(CONFIG_PM_DEVICE_RUNTIME_EXCLUSIVE)
-	if (atomic_sub(&_cpus_active, 1) == 1) {
+	if (atomic_get(&_cpus_active) == 0) {
```

Comment:
TOCTOU ?  N idle threads can get here and have `_cpus_active == 0` and end up all  of them suspending devices. The decrement and check has to be atomic.